code/modules/genetics/machinery/genetics_cloner.dm

```diff
@@ -0,0 +1,738 @@
+turning on the cloner through the control console. The cloner then drains protein from the anchored bidon can, and slowly grows the creature 
+over time. 
+
+Once the creature is grown, the user has to manually vent the grown mob down a disposal pipe into a containment room. If they don't, the creature
+will continue to consume protein and eventually break out of the vat and even start attacking people if it is hostile.
+
+This makes cloning vat is probably the most dangerous tool in Genetics. Because science needs a little danger to be interesting.
+===============================================================================================================================================
+*/
+
+/obj/machinery/genetics/cloner
+	name = "Belvoix Xenofauna Cloning Vat"
+	desc = "A heavily customized cloning vat, retooled for cloning strange and fantastic creatures far and beyond regular fauna. Requires a steady supply of protein to function."
+	icon = 'icons/obj/neotheology_pod.dmi'
+	icon_state = "preview"
+	frame_type = FRAME_VERTICAL
+	circuit = /obj/item/circuitboard/genetics/cloner
+
+	density = TRUE
+	anchored = TRUE
+	layer = 2.8
+	var/obj/machinery/computer/genetics/clone_console/reader
+	var/reader_loc
+
+	var/obj/structure/reagent_dispensers/bidon/container
+	var/container_loc
+
+	var/mob/living/occupant
+
+	var/obj/item/genetics/reject/embryo //Held in the cloner for on-failure events.
+
+	var/obj/item/nonliving_occupant //If or when we're cloning a biological item instead of a mob.
+
+	var/cloning = FALSE //Whether or not the machine is currently attempting to make a clone.
+
+	var/closed = FALSE //Animation marker for closing the vat.
+
+	var/clone_ready = FALSE //If the clone is ready to be expelled
+
+	var/ready_message = FALSE //If the "Clone ready" message has been sent yet
+
+	var/cloning_stage_counter = CLONING_STAGE_BASE
+
+	var/breakout_point = CLONING_BREAKOUT_POINT //The amount of "overgrowing" required to cause a creature to break out of containment.
+
+	var/cloning_speed  = 5	//Try to avoid use of non integer values
+
+	var/embryo_stage = 0 //Which stage the embryo is currently in
+
+	var/feed_the_beast = 0 //1= beast is hungry, 2= beast breaks out. Only increments if the clone is ready and protein is scarce.
+
+	var/progress = 0
+
+	var/progress_increment = 30 SECONDS
+
+	var/progress_benchmark
+
+	//How much protein is eaten in order to advance the progress by the cloning speed.
+	//Higher numbers means the clone eats MORE MEAT.
+	var/protein_consumption = BASE_PROTEIN_CONSUMPTION
+
+	var/datum/genetics/genetics_holder/clone_info //Genetics holder for the mob
+
+	var/datum/genetics/mutation/clone_mutation // The clone mutatation being used to create the mob.
+
+	var/image/anim0 = null
+	var/image/anim1 = null
+
+	var/power_cost = 250
+
+
+	//Disposals info
+	var/datum/gas_mixture/air_contents	// internal reservoir
+	var/mode = 1	// item mode 0=off 1=charging 2=charged
```

Comment:
You could use defines like `OFF`, `CHARGING` and `CHARGED` for the code to be more readable.

Reply:
Fair, I was just copying the disposal code